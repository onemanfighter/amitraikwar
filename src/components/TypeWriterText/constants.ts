import { Speed, SpeedValues } from './type';

const TransformToSpeed: Record<Speed, SpeedValues> = {
  [Speed.slow]: 99,
  [Speed.medium]: 50,
  [Speed.fast]: 10,
};

export { TransformToSpeed };
