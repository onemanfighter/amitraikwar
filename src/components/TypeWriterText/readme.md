# TypeWriterText Component

This component is a simple typewriter effect for text.

## Usage

```tsx
import { TypeWriterText } from '@components';

<TypeWriterText
  fixedText="Hello World"
  textSequence={[
    {
      text: 'Hello',
      time: 1000,
    },
    {
      text: 'World',
      time: 1000,
    },
  ]}
  typeSpeed={50}
  deleteSpeed={50}
/>;
```

## Speed allowed

The speed of the typewriter effect is controlled by the `typeSpeed` prop. The speed is in milliseconds and the default is 50.
Allowed values are between 0 and 100. [1 , 99]

## Sequence of text

The `textSequence` prop allows you to display a sequence of text. The text will be displayed in the order of the array for time mentioned in the array.

```tsx
{
text: 'Hello',
time: 1000
}
```

## Props

| Name         | Type       | Default  | Description                          |
| ------------ | ---------- | -------- | ------------------------------------ |
| fixedText    | string     | required | The text to be displayed             |
| textSequence | sequence[] | required | The text to be displayed in sequence |
| typeSpeed    | number     | 50       | The speed of the typewriter effect   |
| deleteSpeed  | number     | 50       | The speed of the delete effect       |
