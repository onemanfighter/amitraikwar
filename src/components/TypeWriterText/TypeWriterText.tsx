import { TypeAnimation } from 'react-type-animation';
import { Speed, TypeWriterTextProps } from './type';
import { TransformToSpeed } from './constants';

const TypeWriterText = ({
  fixedText,
  textSequence,
  wrapper = 'span',
  typeSpeed = Speed.medium,
  deleteSpeed = Speed.medium,
  styles = { fontSize: '1em', display: 'inline-block' },
}: TypeWriterTextProps) => {
  const sequence = textSequence.flatMap(({ text, wait }) => [
    fixedText + text,
    wait,
  ]);

  return (
    <TypeAnimation
      sequence={sequence}
      wrapper={wrapper}
      speed={TransformToSpeed[typeSpeed]}
      deletionSpeed={TransformToSpeed[deleteSpeed]}
      style={styles}
      repeat={Infinity}
    />
  );
};

export default TypeWriterText;
