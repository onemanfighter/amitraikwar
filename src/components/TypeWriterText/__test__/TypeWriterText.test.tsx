import { render } from '@testing-library/react';
import TypeWriterText from '../TypeWriterText';

jest.mock('react-type-animation', () => ({
  TypeAnimation: ({
    fixedText,
    textSequence,
  }: {
    fixedText: string;
    textSequence: Array<{ text: string; wait: number }>;
  }) => <div>{fixedText}</div>,
}));

describe('TypeWriterText', () => {
  it('should renders correctly', async () => {
    const { container } = render(
      <TypeWriterText
        fixedText="Hello"
        textSequence={[
          {
            text: 'World',
            wait: 2000,
          },
        ]}
      />,
    );

    jest.runAllTimersAsync();

    expect(container).toMatchSnapshot();
  });

  it('should renders correctly with empty text', async () => {
    const { container } = render(
      <TypeWriterText fixedText="" textSequence={[]} />,
    );

    jest.runAllTimersAsync();

    expect(container).toMatchSnapshot();
  });

  it('should renders correctly with empty textSequence', async () => {
    const { container } = render(
      <TypeWriterText fixedText="Hello" textSequence={[]} />,
    );

    jest.runAllTimersAsync();

    expect(container).toMatchSnapshot();
  });
});
