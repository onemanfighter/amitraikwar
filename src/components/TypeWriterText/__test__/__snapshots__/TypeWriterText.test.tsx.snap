// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeWriterText should renders correctly 1`] = `
<div>
  <div />
</div>
`;

exports[`TypeWriterText should renders correctly with empty text 1`] = `
<div>
  <div />
</div>
`;

exports[`TypeWriterText should renders correctly with empty textSequence 1`] = `
<div>
  <div />
</div>
`;
